File: backend/app/schemas/live_session.py
from datetime import datetime
from typing import Literal

from app.schemas.base import CamelModel

MarkType = Literal["perfect", "hesitation", "mistake"]


class LiveSessionOut(CamelModel):
    id: str
    class_id: str
    class_name: str
    teacher_name: str
    status: str
    started_at: datetime
    ended_at: datetime | None


class LiveSessionParticipantOut(CamelModel):
    student_id: str
    student_name: str
    joined_at: datetime
    left_at: datetime | None
    duration_seconds: float | None


class LiveSessionMistakeOut(CamelModel):
    student_id: str
    student_name: str
    mark_type: MarkType
    note: str | None
    created_at: datetime


class LiveSessionReportOut(CamelModel):
    session: LiveSessionOut
    participants: list[LiveSessionParticipantOut]
    mistakes: list[LiveSessionMistakeOut]


class StartLiveSessionRequest(CamelModel):
    class_id: str

File: src/types/liveSession.ts
export type LiveSessionStatus = 'scheduled' | 'live' | 'ended';
export type MarkType = 'perfect' | 'hesitation' | 'mistake';
export type PeerRole = 'teacher' | 'student';

export interface LiveSession {
  id: string;
  classId: string;
  className: string;
  teacherName: string;
  status: LiveSessionStatus;
  startedAt: string;
  endedAt: string | null;
}

export interface LiveSessionParticipant {
  studentId: string;
  studentName: string;
  joinedAt: string;
  leftAt: string | null;
  durationSeconds: number | null;
}

export interface LiveSessionMistake {
  studentId: string;
  studentName: string;
  markType: MarkType;
  note: string | null;
  createdAt: string;
}

export interface LiveSessionReport {
  session: LiveSession;
  participants: LiveSessionParticipant[];
  mistakes: LiveSessionMistake[];
}

export interface RemotePeer {
  peerId: string;
  role: PeerRole;
  name: string;
  connectionState: RTCPeerConnectionState;
}

export interface LiveMistakeEvent {
  studentPeerId: string;
  markType: MarkType;
  note: string | null;
  at: number;
}
